resources/inspec_gem.rb

```diff
@@ -0,0 +1,41 @@
+provides :inspec_gem
+resource_name :inspec_gem
+
+property :gem_name, String, name_property: true
+property :version, String
+property :source, String
+
+default_action :install
+
+action :install do
+  if !version.nil?
+    converge_by "install InSpec version #{version}" do
```

Comment:
I'm partially against that since we should be trusting chef (and chef_gem) to do what we tell it to. But I'm all for setting us up for greater success... to your next comment about removing-then-installing, I think that's a fine idea and will test that.